### ISSUE TITLE
behavior_path_planner can't plan a path with the first given goal command

### ISSUE THREAD

**Issue by mehmetdogru** [autowarefoundation/autoware.universe#636]
### Checklist

- [X] I've read the [contribution guidelines](https://github.com/autowarefoundation/autoware/blob/main/CONTRIBUTING.md).
- [X] I've searched other issues and no duplicate issues were found.
- [X] I'm convinced that this is not my fault but a bug.

### Description

Most of the time, behavior_path_planner sometimes can't plan a path with the first given goal command from rviz2. And the node gives this error:
```
[component_container_mt-13] [ERROR] [1649165181.816988526] [planning.scenario_planning.lane_driving.behavior_planning.behavior_path_planner]: behavior path output is empty! Stop publish.
```
And when the second time a goal command is given (even though exactly the same goal pose as the first one) behavior_path_planner is able to plan the path.

### Expected behavior

behavior_path_planner should be able to calculate a path with the first goal pose given.

### Actual behavior

Most of the time, behavior_path_planner sometimes can't plan a path with the first given goal command from rviz2

### Steps to reproduce

1. Launch planning_simulator.launch.xml with any lanelet2 map and any .pcd file as it is described [here](https://autowarefoundation.github.io/autoware-documentation/main/tutorials/ad-hoc-simulation/planning-simulation/#:~:text=%23-,Launch%20Autoware.,-source%20~/autoware/install).
2. Then [set an initial pose for the ego vehicle](https://autowarefoundation.github.io/autoware-documentation/main/tutorials/ad-hoc-simulation/planning-simulation/#:~:text=Set%20a%20goal%20pose%20for%20the%20ego%20vehicle.).
3. [Set a goal pose for the ego vehicle](https://autowarefoundation.github.io/autoware-documentation/main/tutorials/ad-hoc-simulation/planning-simulation/#:~:text=Set%20a%20goal%20pose%20for%20the%20ego%20vehicle).
4. Repeat the 3rd step and observe if the path is created by behavior_path_planner.

### Versions

- Autoware.universe commit hash: 70bc967d22a6bb8ca49c55296de833261a7f63fc

**Comment by maxime-clem** [comment]
I am able to reproduce the issue but it is inconsistent and setting a goal sometimes works the first time.

Using `git bisect` I have identified the first bad commit as 982263a78e1079f054b428818636de1287801109.

**Comment by maxime-clem** [comment]
This issue was fixed by c341d4c4e4a79dd0b1f2259653a4f6cc3f7e4c50 but I observed a new issue where `behavior_planning` crashes after setting a goal.
```
[component_container_mt-13] Inconsistency detected by ld.so: ../elf/dl-tls.c: 481: _dl_allocate_tls_init: Assertion `listp->slotinfo[cnt].gen <= GL(dl_tls_generation)' failed!
[ERROR] [component_container_mt-13]: process has died [pid 9361, exit code 127, cmd '/opt/ros/galactic/lib/rclcpp_components/component_container_mt --ros-args -r __node:=behavior_planning_container -r __ns:=/planning/scenario_planning/lane_driving/behavior_planning --params-file /tmp/launch_params_bzfw7nao'].
```
